src/commander.js

```diff
@@ -0,0 +1,348 @@
+                setTimeout(function () {
+                    // open add question menu after delay to allow atwo menu
+                    // to hide. https://stackoverflow.com/a/29572644/10840
+                    $(".fd-add-question-dropdown").addClass('open');
+                }, 0);
+            });
+            $(".fd-add-question-dropdown").append(cmd.container.hide());
+        },
+        getToolsMenuItems: function () {
+            var menuItems = this.__callOld();
+            menuItems.splice(menuItems.length - 1, 0, {
+                name: gettext('Command Bar'),
+                icon: "fa fa-terminal",
+                hotkey: "Ctrl+;",
+                action: function () { showCommander(this.data.commander); },
+            });
+            return menuItems;
+        },
+    });
+
+    function showCommander(cmd) {
+        setupAutocomplete(cmd);
+        cmd.addQuestionButton.hide();
+        cmd.container.show();
+        cmd.input.focus().select();
+    }
+
+    function hideCommander(cmd) {
+        cmd.addQuestionButton.show();
+        cmd.container.hide();
+        cmd.input.val("").removeClass("alert-danger");
+    }
+
+    function setupAutocomplete(cmd) {
+        if (cmd.atwhoConfig) {
+            return;
+        }
+        _.extend(cmd, configure(cmd.vellum));
+        cmd.input.atwho(cmd.atwhoConfig);
+        atwho.autocomplete(
+            cmd.input,
+            {form: cmd.vellum.data.core.form, on: _.identity},  // fake mug
+            {useHashtags: true, tabSelectsMatch: true}          // options
+        );
+        cmd.input.on("inserted.atwho", function (event) {
+            event.preventDefault();
+        });
+    }
+
+    function configure(vellum) {
+        function getMugClassName(typeName) {
+            typeName = typeName.toLowerCase();
+            if (mugTypes.hasOwnProperty(typeName)) {
+                return mugTypes[typeName].__className;
+            }
+        }
+
+        function getPosition(position) {
+            if (position) {
+                position = position.toLowerCase();
+                if (positionMap.hasOwnProperty(position)) {
+                    position = positionMap[position];
+                }
+            }
+            return position;
+        }
+
+        function getMug(path) {
+            var refMug;
+            if (path) {
+                refMug = vellum.getMugByPath(path);
+                if (!refMug) {
+                    throw new Error("bad path: " + path);
+                }
+            } else {
+                refMug = vellum.getCurrentlySelectedMug();
+            }
+            return refMug;
+        }
+
+        function literals(names, getArg) {
+            if (!getArg) {
+                throw new Error("getArg parameter is required");
+            }
+            var source = _.map(names, RegExp.escape).join("|");
+            return {regexp: source, names: names, getArg: getArg};
+        }
+
+        function xname(predicate) {
```

Comment:
I'm not exactly sure what this or itemizer do or why they do it. Could you add some comments?

Also xname is only used once. Are you planning on using it more later? I think it could be simplified by referencing itemizer directly. here